src/Processors/Formats/Impl/PrettyBlockOutputFormat.h

```diff
@@ -73,10 +73,22 @@ void registerPrettyFormatWithNoEscapesAndMonoBlock(FormatFactory & factory, cons
             const Block & sample,
             const FormatSettings & format_settings)
         {
-            if (no_escapes)
+            FormatSettings changed_settings = format_settings;
+            using PrettyColor = FormatSettings::Pretty::PrettyColor;
+            switch (format_settings.pretty.output_format_pretty_color)
+            {
+                case PrettyColor::Off:
+                    changed_settings.pretty.color = false;
+                    break;
+                case PrettyColor::On:
+                    changed_settings.pretty.color = no_escapes ? false : true; 
+                    break;
+                case PrettyColor::Auto:
+                    changed_settings.pretty.color = isatty(STDOUT_FILENO) && no_escapes ? false : true; 
```

Comment:
We will output data into our `WriteBuffer`, not always to stdout. So, we should first check if our buffer will output data into stdout and then use `isatty` function on `STDOUT_FILENO`. We can create a separate function for it:
```
bool isWritingToTerminal(const WriteBuffer & buf)
{
    const auto * write_buffer_to_descriptor = typeid_cast<const WriteBufferFromFileDescriptor *>(&buf);
    return write_buffer_to_descriptor && write_buffer_to_descriptor->getFD() == STDOUT_FILENO && isatty(STDOUT_FILENO);
}
```

We can place this function in file `src/IO/WriteHelpers.h`.

Also if I understood correctly, we should do `|| no_escapes` here, not `&& no_escapes`. So it will look like this:
```
                    changed_settings.pretty.color = isWritingToTerminal(buf) || no_escapes ? false : true; 
```